cumulusci/robotframework/Salesforce.py

```diff
@@ -0,0 +1,477 @@
+            except SalesforceResourceNotFound:
+                self.builtin.log('    {type} {id} is already deleted'.format(**record))
+            except SalesforceMalformedRequest as e:
+                self.builtin.log('    {type} {id} could not be deleted:'.format(**record), level='WARN')
+                self.builtin.log('      {}'.format(e), level='WARN')
+        
+    def get_current_record_id(self):
+        """ Parses the current url to get the object id of the current record.
+            Expects url format like: [a-zA-Z0-9]{15,18}
+        """
+        url = self.selenium.get_location()
+        for part in url.split('/'):
+            if re.match(OID_REGEX, part):
+                return part
+        raise AssertionError("Could not parse record id from url: {}".format(url))
+
+    def get_locator(self, path, *args, **kwargs):
+        """ Returns a rendered locator string from the Salesforce locators
+            dictionary.  This can be useful if you want to use an element in
+            a different way than the built in keywords allow.
+        """
+        locator = locators
+        for key in path.split('.'):
+            locator = locator[key]
+        return locator.format(*args, **kwargs)
+
+    def get_record_type_id(self, obj_type, developer_name):
+        soql = "SELECT Id FROM RecordType WHERE SObjectType='{}' and DeveloperName='{}'".format(
+            obj_type,
+            developer_name,
+        )
+        res = self.cumulusci.sf.query_all(soql)
+        return res['records'][0]['Id']
+
+    def get_related_list_count(self, heading):
+        locator = locators['record']['related']['count'].format(heading)
+        return self._call_selenium('_get_related_list_count', True, locator)
+
+    def _get_related_list_count(self, locator):
+        count = self.selenium.get_webelement(locator).text
+        count = count.replace('(','').replace(')','')
+        return int(count)
+        
+    def go_to_object_home(self, obj_name):
+        """ Navigates to the Home view of a Salesforce Object """
+        url = self.cumulusci.org.lightning_base_url
+        url = '{}/one/one.app#/sObject/{}/home'.format(url, obj_name)
+        self.selenium.go_to(url)
+        time.sleep(5)
+        self._wait_until_loading_is_complete()
+    
+    def go_to_object_list(self, obj_name, filter_name=None):
+        """ Navigates to the Home view of a Salesforce Object """
+        url = self.cumulusci.org.lightning_base_url
+        url = '{}/one/one.app#/sObject/{}/list'.format(url, obj_name)
+        if filter_name:
+            url += '?filterName={}'.format(filter_name)
+        self.selenium.go_to(url)
+        time.sleep(5)
+        self._wait_until_loading_is_complete()
+
+    def go_to_record_home(self, obj_id, filter_name=None):
+        """ Navigates to the Home view of a Salesforce Object """
+        url = self.cumulusci.org.lightning_base_url
+        url = '{}/one/one.app#/sObject/{}/view'.format(url, obj_id)
```

Comment:
can we pull all the different URL strings we're using into module constants? url v2 compatibilit is also something we need to think about.

Reply:
I've been thinking about having a library init option for LEX url format version and handling it that way.  We'd need to build up a test suite that could validate that.